File: 二叉树/102.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

# BFS
class Solution:
    def levelOrder(self, root: TreeNode):
        queue = []
        res = []
        if(root):
            queue.append(root)
        while(queue):
            temp = []
            length = len(queue)
            for i in range(length):
                node = queue.pop(0)
                temp.append(node.val)
                if(node.left):
                    queue.append(node.left)
                if(node.right):
                    queue.append(node.right)
            res.append(temp)
        return res

root = TreeNode(3)
root.left = TreeNode(9)
root.right = TreeNode(20)
root.right.left = TreeNode(15)
root.right.right = TreeNode(7)

obj = Solution()
print(obj.levelOrder(root))

File: 二叉树/102.cpp
#include <vector>
#include <queue>
using namespace std;

//Definition for a binary tree node.
struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution
{
public:
    vector<vector<int>> levelOrder(TreeNode *root)
    {
        queue<TreeNode *> NodeList;
        vector<vector<int>> result;

        int layer_length;
        if (root)
            NodeList.push(root);
        while (!NodeList.empty())
        {
            vector<int> layer;
            layer_length = NodeList.size();
            for (int i = 0; i < layer_length; i++)
            {
                TreeNode *Node = NodeList.front();
                NodeList.pop();
                layer.push_back(Node->val);
                if (Node->left)
                    NodeList.push(Node->left);
                if (Node->right)
                    NodeList.push(Node->right);
            }
            result.push_back(layer);
        }
        return result;
    }
};